### PR TITLE
[controller] Superset schema generation should reflect default value updates in new value schema

### PR DESCRIPTION
**Pull request by sixpluszero**
## [controller] Superset schema generation should reflect default value updates in new value schema
When user updates a field default value, we should reflect it in the changes in superset schema, which is used in read-compute / partial update store. This should apply to all the value type, including union schema. 

## How was this PR tested?
Add a new integration test to validate
## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java [hunk 1]
```diff
@@ -34,56 +34,56 @@ public static boolean isSupersetSchema(Schema s1, Schema s2) {
...
-   * @param existingSchema schema existing in the repo
-   * @param newSchema schema to be added.
+   * schemas and the generated schema will pick the default value from new value schema.
+   * @param newSchema schema existing in the repo
+   * @param oldSchema schema to be added.
    * @return super-set schema of existingSchema abd newSchema
    */
-  public static Schema generateSuperSetSchema(Schema existingSchema, Schema newSchema) {
-    if (existingSchema.getType() != newSchema.getType()) {
+  public static Schema generateSuperSetSchema(Schema newSchema, Schema oldSchema) {
```
Comment (m-nagarajan) [inline internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java:42]: I feel not changing the function definition should make the diff a bit concise. what do you think?

Comment (sixpluszero) [reply internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java:42]: Do you mean reverse the order back? Let me see if I can achieve that.

Comment (sixpluszero) [reply internal/venice-client-common/src/main/java/com/linkedin/venice/utils/AvroSupersetSchemaUtils.java]: Fixed. Could you check again?

**Review by m-nagarajan — COMMENTED** [review]
LGTM Thanks @sixpluszero, left one minor comment.

**Review by m-nagarajan — APPROVED** [review]
Thanks @sixpluszero

**Pull request merged**